src/includes/Helpers/PaymentHelper.php

```diff
@@ -116,11 +138,24 @@ public function __construct() {
 	 */
 	public function handle_ipn_callback() {
 		$payment_id = $this->get_payment_to_validate();
+
+		try {
+			$this->security_helper->validate_ipn_signature(
+				$payment_id,
+				$this->alma_settings->get_active_api_key(),
+				$_SERVER['HTTP_X_ALMA_SIGNATURE']
```

Comment:
Add verification if key HTTP_X_ALMA_SIGNATURE doesn't exist to avoid error and throw an exception instead

Reply:
Done